src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -130,10 +130,10 @@ public bool IsDiagnosticSuppressed(Diagnostic diagnostic, out AttributeData supp
             return false;
         }
 
-        private bool IsDiagnosticSuppressed(Diagnostic diagnostic, out SuppressMessageInfo info)
-            => IsDiagnosticSuppressed(diagnostic.Id, diagnostic.Location, out info);
+        private bool IsDiagnosticSuppressed(Diagnostic diagnostic, Func<SyntaxTree, SemanticModel> getSemanticModel, out SuppressMessageInfo info)
+            => IsDiagnosticSuppressed(diagnostic.Id, diagnostic.Location, getSemanticModel, out info);
 
-        private bool IsDiagnosticSuppressed(string id, Location location, out SuppressMessageInfo info)
+        private bool IsDiagnosticSuppressed(string id, Location location, Func<SyntaxTree, SemanticModel> getSemanticModel, out SuppressMessageInfo info)
```

Comment:
>Func<SyntaxTree, SemanticModel> [](start = 74, length = 31)

Consider using `Func<Compilation, SyntaxTree, SemanticModel>` to avoid unnecessary closure classes in the callers.

Reply:
Thanks, will fix.